python/educational-core/student/src/com/jetbrains/edu/learning/stepic/EduStepicConnector.java

```diff
@@ -385,32 +417,106 @@ public static Course getCourse(@NotNull final Project project, @NotNull final Co
-      if (step.options.test != null) {
-        LOG.warn("step.o = " + step.toString());
-        for (StepicWrappers.TestFileWrapper wrapper : step.options.test) {
-          task.addTestsTexts(wrapper.name, wrapper.text);
+      if (supported(step.name)) {
+        final Task task = new Task();
+        task.setStepicId(stepicIds.get(i++));
+
+        switch (step.name) {
+          case (CODE_PREFIX):
+            createCodeTask(task, step);
+            break;
+          case (PYCHARM_PREFIX):
+            createPyCharmTask(task, step);
+            break;
         }
+        lesson.taskList.add(task);
+      }
+    }
+  }
+
+  private static boolean supported(String name) {
+    return CODE_PREFIX.equals(name) || PYCHARM_PREFIX.equals(name);
+  }
+
+  private static void createPyCharmTask(Task task, StepicWrappers.Step step) {
+    task.setName(step.options != null ? step.options.title : PYCHARM_PREFIX);
+    task.setText(step.text);
+    for (StepicWrappers.TestFileWrapper wrapper : step.options.test) {
+      task.addTestsTexts(wrapper.name, wrapper.text);
+    }
+
+    task.taskFiles = new HashMap<String, TaskFile>();      // TODO: it looks like we don't need taskFiles as map anymore
+    if (step.options.files != null) {
+      for (TaskFile taskFile : step.options.files) {
+        task.taskFiles.put(taskFile.name, taskFile);
       }
+    }
+  }
 
-      task.taskFiles = new HashMap<String, TaskFile>();      // TODO: it looks like we don't need taskFiles as map anymore
-      if (step.options.files != null) {
-        for (TaskFile taskFile : step.options.files) {
-          task.taskFiles.put(taskFile.name, taskFile);
+  private static void createCodeTask(Task task, StepicWrappers.Step step) {
+    task.setName("step" + task.getStepicId());
+    if (step.options.samples != null) {
+      final StringBuilder builder = new StringBuilder();
+      for (List<String> sample : step.options.samples) {
+        if (sample.size() == 2) {
+          builder.append("<b>Sample Input:</b><br>");
+          builder.append(StringUtil.replace(sample.get(0), "\n", "<br>"));
+          builder.append("<br>");
+          builder.append("<b>Sample Output:</b><br>");
+          builder.append(StringUtil.replace(sample.get(1), "\n", "<br>"));
+          builder.append("<br><br>");
         }
       }
-      lesson.taskList.add(task);
+      task.setText(step.text + "<br>" + builder.toString());
+    }
+
+    if (step.options.executionMemoryLimit != null && step.options.executionTimeLimit != null) {
+      String builder = "<b>Memory limit</b>: " +
+                       step.options.executionMemoryLimit + " Mb" +
+                       "<br>" +
+                       "<b>Time limit</b>: " +
+                       step.options.executionTimeLimit + "s" +
+                       "<br><br>";
+      task.setText(task.getText() + builder);
     }
+
+    final TaskFile taskFile = new TaskFile();
+    taskFile.name = "Main.java";
```

Comment:
Stepic connector is in language independent part so you can not use java stuff here